src/components/pill/pill.style.js

```diff
@@ -99,20 +89,196 @@ const PillStyle = styled.span`
+              margin-top: -1px;
+
+              &:before {
+                font-size: 10px;
+              }
+            }
+          }
+        `}
+
+        ${size === 'L' && css`
+          padding: 0 36px 0 20px;
+          border-radius: 13px;
+
+          button {
+            width: 28px;
+            padding: 0;
+            border-radius: 0 11px 11px 0;
+
+            .carbon-icon {
+              padding: 3px 8px 4px 8px;
+              margin-top: -1px;
+
+              &:before {
+                font-size: 12px;
+              }
+            }
+          }
+        `}
+
+        ${size === 'XL' && css`
+          padding: 0 40px 0 24px;
+          border-radius: 15px;
+
+          button {
+            width: 32px;
+            padding: 0;
+            border-radius: 0 14px 14px 0;
+
+            .carbon-icon {
+              padding: 3px 9px 5px 9px;
+              margin-top: -1px;
+
+              &:before {
+                font-size: 13px;
+              }
+            }
+          }
+        `}
+      `}
+
+      ${!isClassic(theme, colorVariant) && !isDeletable && css`
+        ${size === 'S' && css`
+          padding: 0 12px;
+
+          button {
+            padding: 0;
+
+            .carbon-icon {
+              padding: 0;
+
+              &:before {
+                font-size: 7px;
+              }
+            }
+          }
+        `}
+
+        ${size === 'M' && css`
+          padding: 0 16px;
+          border-radius: 12px;
+
+          button {
+            width: 24px;
+            padding: 0;
+            border-radius: 0 8px 8px 0;
+
+            .carbon-icon {
+              padding: 2px 7px 3px 7px;
+              margin-top: -1px;
+
+              &:before {
+                font-size: 10px;
+              }
             }
-            &:before {
-              font-size: 12px;
+          }
+        `}
+
+        ${size === 'L' && css`
+          padding: 0 20px;
+          border-radius: 13px;
+
+          button {
+            width: 28px;
+            padding: 0;
+            border-radius: 0 10px 10px 0;
+
+            .carbon-icon {
+              padding: 3px 8px 4px 8px;
+              margin-top: -1px;
+
+              &:before {
+                font-size: 12px;
+              }
             }
           }
+        `}
+
+        ${size === 'XL' && css`
+          padding: 0 24px;
+          border-radius: 15px;
+
+          button {
+            width: 32px;
+            padding: 0;
+            border-radius: 0 12px 12px 0;
+
+            .carbon-icon {
+              padding: 3px 9px 5px 9px;
+              margin-top: -1px;
+
+              &:before {
+                font-size: 13px;
+              }
+            }
+          }
+        `}
       `}
 
-      ${isClassic(theme, colorVariant) && classicThemeForPill(colorVariant, inFill, isDeletable)}
+      ${isClassic(theme, colorVariant) && classicThemeForPill(colorVariant, inFill, isDeletable, size)}
+
+      ${size === 'S' && css`
```

Comment:
Are all the styles on lines 259-281 the classic styles? If so these should go in `src/components/pill/pill-classic.style.js` (the call to `classicThemeForPill` above would then return the styles from that file if the theme is classic.

With this file written as it is at the moment, all of the styles appearing below line 257 will override the styles defined in `src/components/pill/pill-classic.style.js`. The call to import the classic styles should always be at the bottom of your styled components file so that it overrides the DLS styles defined above it in this file.